File: ios/NfcEventEmitter.m
//
 //  NfcEventEmitter.m
 //  react-native-new-ton-nfc-card-lib
 //
 //  Created by Alina Alinovna on 09.12.2020.
 //

 #import <Foundation/Foundation.h>
 #import <React/RCTBridgeModule.h>
 #import <React/RCTEventEmitter.h>

 @interface RCT_EXTERN_MODULE(NfcEventEmitter, RCTEventEmitter)

 RCT_EXTERN_METHOD(supportedEvents)

 @end

File: ios/NfcEventEmitter.swift

//
 //  NfcEventEmitter.swift
 //  react-native-new-ton-nfc-card-lib
 //
 //  Created by Alina Alinovna on 09.12.2020.
 //

 import Foundation

 @objc(NfcEventEmitter)
 class NfcEventEmitter: RCTEventEmitter {
     static let NFC_TAG_CONNECTED_EVENT:String = "nfcTagConnected"

     public static var emitter:RCTEventEmitter!

     override init() {
         super.init()
         NfcEventEmitter.emitter = self
     }

     override func supportedEvents() -> [String]! {
         return [NfcEventEmitter.NFC_TAG_CONNECTED_EVENT]
     }
 }
